src/main/python/pybuilder/errors.py

```diff
@@ -47,14 +47,8 @@ def __init__(self, name):
 
 
 class CircularTaskDependencyException(PyBuilderException):
-    def __init__(self, first, second=None, message=None):
```

Comment:
This fix doesn't look right at all. The code is valid and is used in unit tests. I'll dive deeper into this.

Reply:
Bad joke

Reply:
We have two places where this Exception is used: 
https://github.com/pybuilder/pybuilder/blob/7648c1ed97b5f77a20d4d0a3cb984cf842541b79/src/main/python/pybuilder/execution.py#L397
and 
https://github.com/pybuilder/pybuilder/blob/7648c1ed97b5f77a20d4d0a3cb984cf842541b79/src/main/python/pybuilder/execution.py#L419
And called with only first argument. So 
```
class CircularTaskDependencyException(PyBuilderException):
    def __init__(self, first, second=None, message=None):
        if message:
            super(CircularTaskDependencyException, self).__init__(message)
        elif second:
            super(CircularTaskDependencyException, self).__init__("Circular task dependency detected between %s and %s",
                                                                  first, second)
            self.first = first
            self.second = second
``` just passed without any action
Maybe was planned next:
```
class CircularTaskDependencyException(PyBuilderException):
    def __init__(self, message=None, first=None, second=None):
```
it makes more sence for this `if`, but in any cases into code we use only message and second part (with first/second) could be removed as not used.

Reply:
And I totally agree- it's really sad and big problem that tests passed unexpectedly. :((((

Reply:
I found this problem as result of Pyb work - #498

Reply:
For myself: check this situation on pytest instead of unittest